lib/Runtime/Base/FunctionBody.h

```diff
@@ -2690,6 +2693,22 @@ namespace Js
         bool GetHasRestParameter() const { return (flags & Flags_HasRestParameter) != 0; }
         void SetHasRestParameter() { SetFlags(true, Flags_HasRestParameter); }
 
+        bool NeedScopeObjectForArguments(bool hasNonSimpleParams)
+        {
+            Assert(HasReferenceableBuiltInArguments());
+            // We can avoid creating a scope object with arguments present if:
+            bool dontNeedScopeObject =
+                // Either we are in strict mode, or have strict mode formal semantics from a non-simple parameter list, and
+                (GetIsStrictMode() || hasNonSimpleParams)
+                // Neither of the scopes are objects
+                && !HasScopeObject();
+            
+            return 
+                // Regardless of the conditions above, we won't need a scope object if there aren't any formals.
+                (GetInParamsCount() > 1 || GetHasRestParameter())
```

Comment:
Rest parameter makes param to non-simple param already. Do you have redundant check here?